tests/modules/programs/ssh/match-blocks-match-and-hosts.nix

```diff
@@ -0,0 +1,32 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.ssh = {
+      enable = true;
+      matchBlocks = {
+        abc = { port = 2222; };
+
+        xyz = {
+          match = "host xyz canonical";
+          port = 2223;
+        };
+
+        "* !github.com" = { port = 516; };
+      };
+    };
+
+    home.file.assertions.text = builtins.toJSON
+      (map (a: a.message) (filter (a: !a.assertion) config.assertions));
```

Comment:
Are you sure this is necessary?

Reply:
honestly, I don't know. I've never written any tests like this before. This is a _inspiraton_ (read: _copy-paste_) from another test which I adapted to the best of my understanding